### PR TITLE
Feat: Products pages

### PR DESCRIPTION
**Pull request by abdul-moize**
This branch adds the Products template to main branch

To test this PR. follow these steps

1. Pull branch and run: python manage.py migrate && python manage.py runserver. also create a super user by python manage.py createsuper.
2. Pull this branch & run: npm start
3. Open http://localhost:3000/ in browser

Browse the site and let me if there are any issues, comments & concerns

### CODE REVIEW

File: src/services/cartService.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    headers: {
+      Authorization: "Token " + localStorage.getItem("token"),
+    },
+    body: formData,
+  }).then((res) => {
+    console.log(res);
+    if (res.status === 200) return res.json();
+    throw new OutOfStockException("Product is out of stock");
+  });
+}
```
Comment (srehaider) [inline src/services/cartService.js:18]: You can't throw `OutOfStockException` exception when the response status code is not 200, there can be other possible errors too like 500 and so on. So, handle this case specifically and all other possible cases as well.

Comment (abdul-moize) [reply src/services/cartService.js]: Done

File: src/pages/addProduct/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          }
+        });
+      });
+  }
+  return (
+    <div
+      className="form-container"
+      style={{ background: "url(/background_banner.jpg)" }}
+    >
+      <div className="title-field">Add Product</div>
```
[the remark this answers is not in the document]
Comment (abdul-moize) [reply src/pages/addProduct/index.js]: Done

File: src/pages/addProduct/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  const history = useHistory();
+  const [nameError, setNameError] = useState("");
+  const formRef = useRef();
+  if (!isContentManager()) {
+    history.replace(HOMEPAGE);
+  }
+  function submitForm(event) {
+    event.preventDefault();
+    const formData = new FormData(formRef.current);
+    addProduct(formData)
```
[the remark this answers is not in the document]
Comment (abdul-moize) [reply src/pages/addProduct/index.js]: Done

File: src/pages/addProduct/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import { isContentManager } from "../../utils";
+import "./addProduct.css";
+
+export function AddProduct() {
+  const history = useHistory();
+  const [nameError, setNameError] = useState("");
+  const formRef = useRef();
+  if (!isContentManager()) {
+    history.replace(HOMEPAGE);
+  }
```
[the remark this answers is not in the document]
Comment (abdul-moize) [reply src/pages/addProduct/index.js]: Done

File: src/pages/addProduct/addProduct.css [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  height: 50px;
+  font-size: 25px;
+}
+
+.labels{
+  display: flex;
+  flex-direction: column;
+  justify-content: left;
+  width: 100%;
+}
```
[the remark this answers is not in the document]
Comment (abdul-moize) [reply src/pages/addProduct/addProduct.css]: I have changed their names to better represent and removed some of these as they are not required.

File: src/pages/addProduct/addProduct.css [hunk 6]
```diff
@@ -0,0 +1,104 @@
+.form-container{
```
[the remark this answers is not in the document]
Comment (abdul-moize) [reply src/pages/addProduct/addProduct.css]: Done

File: src/pages/addProduct/index.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import { useRef, useState } from "react";
+import { useHistory } from "react-router";
+import { useEffect } from "react/cjs/react.development";
```
Comment (srehaider) [inline src/pages/addProduct/index.js:3]: import it directly from react.

File: src/pages/addProduct/index.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+export function AddProduct() {
+  const history = useHistory();
+  const [nameError, setNameError] = useState("");
+  const formRef = useRef();
+  useEffect(() => {
+    if (!isContentManager()) {
+      history.replace(HOMEPAGE);
+    }
+  });
```
Comment (srehaider) [inline src/pages/addProduct/index.js:18]: When you don't pass dependency array, it means callback will be executed on each render which is not intentional behavior, pass an empty array as a dependency array to execute callback only on the first render. Leave blank lines before and after this code block.
```suggestion
  useEffect(() => {
    if (!isContentManager()) {
      history.replace(HOMEPAGE);
    }
  }, []);
```

File: src/services/productService.js [hunk 9]
```diff
@@ -14,3 +18,23 @@ export default function getProducts() {
...
+    throw res.json();
+  });
+}
+
+export function getProduct(id) {
+  return fetch(`${PRODUCT_API}${id}`).then((res) => {
+    if (res.status === 200) return res.json();
+    throw new DoesNotExistException("Product does not exist");
+  });
+}
```
Comment (srehaider) [inline src/services/productService.js:40]: Handle errors here properly, to keep services consistent, either you receive an error message or you receive data.

**Pull request merged**